utils/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/AvroDefaultExpressionExtractor.scala

```diff
@@ -0,0 +1,117 @@
+package pl.touk.nussknacker.engine.avro
+
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.ValidatedNel
+import org.apache.avro.{JsonProperties, LogicalTypes, Schema}
+import pl.touk.nussknacker.engine.graph.expression.Expression
+
+import scala.annotation.tailrec
+import scala.reflect.ClassTag
+
+
+object AvroDefaultExpressionExtractor {
+
+  sealed trait AvroDefaultToSpELExpressionError extends Exception
+
+  case object NullNotAllowed extends AvroDefaultToSpELExpressionError {
+    override val getMessage: String = "Value is not nullable"
+  }
+
+  case object InvalidValue extends AvroDefaultToSpELExpressionError {
+    override val getMessage: String = "Value is invalid"
+  }
+
+  case class TypeNotSupported(schema: Schema) extends AvroDefaultToSpELExpressionError {
+    override def getMessage: String = s"Default value for ${schema.getType} is not supported."
+  }
+}
+
+/**
+  *
+  * @param handleNotSupported when true values of not supported types are returned as valid None, otherwise invalid TypeNotSupported is returned
+  */
+class AvroDefaultExpressionExtractor(fieldSchema: Schema.Field, handleNotSupported: Boolean) {
+  import AvroDefaultExpressionExtractor._
+  import scala.collection.JavaConverters._
+  import pl.touk.nussknacker.engine.spel.Implicits.asSpelExpression
+
+  private val validatedNullExpression: ValidatedNel[AvroDefaultToSpELExpressionError, Option[Expression]] =
+    Valid(Some(asSpelExpression("null")))
+
+  private val typeNotSupported: ValidatedNel[AvroDefaultToSpELExpressionError, Option[Expression]] =
+    if (handleNotSupported)
+      Valid(None)
+    else
+      Invalid(TypeNotSupported(fieldSchema.schema())).toValidatedNel
+
+  def toExpression: ValidatedNel[AvroDefaultToSpELExpressionError, Option[Expression]] =
+    if (fieldSchema.hasDefaultValue)
+      toExpression(fieldSchema.schema(), fieldSchema.defaultVal())
+    else
+      Valid(None)
+
+  /**
+    * The expression has to correspond to type extracted by AvroSchemaTypeDefinitionExtractor#typeDefinition
+    * !when applying changes keep in mind that this Schema.Type pattern matching is duplicated in AvroSchemaTypeDefinitionExtractor
+    */
+  @tailrec
+  private def toExpression(schema: Schema, defaultValue: AnyRef): ValidatedNel[AvroDefaultToSpELExpressionError, Option[Expression]] = {
```

Comment:
I've added support of UUID,
as far as Byte type is concerned, we are typing it as ByteBuffer, and I'm not sure if we want to present such a SpEL expression:
`T(java.nio.ByteBuffer).wrap(str.toString.getBytes(T(java.nio.charset.StandardCharsets.UTF_8)))`

as far as date/time types are concerned: LocalDate, LocalTime, Instant - it might be a bit tricky, because of the time zone.
We can relay on system default time zone, or utc?

wdyt?

Reply:
We support avro 1.9 (https://avro.apache.org/docs/1.9.0/spec.html) - it doesn't support LocalDate. For timestamp you don't need to handle time zone - it is just `Instant.ofEpochMilli`, for `timeMillis` it isn't `LocalTime.ofNanoOfDay`? Regards to bytes - they are represented as hex, take a look at docs: `"\u00FF"` but it is a little bit harder to parse it in one liner: https://stackoverflow.com/questions/140131/convert-a-string-representation-of-a-hex-dump-to-a-byte-array-using-java Please add TODO and we can do it later

Reply:
I've added support for timestamp-millis.
As far as other types are concerned I'm not sure If we'd like to handle them, eg the bytes case where the expression might look odd and repulsive.
It's up to you - If you think we should handle it some how in the futre I'll add TODO and related Issue.